juju/juju
# Provisioner state address

The provisioner worker was calling a StateAddresses() API but it doesn't need that info. Remove the API and fix the associated fallout.

The second commit is a drive by to move some trivial container helper methods from state to core. This removes one of the cases where the worker package imports state. Sadly it's one of many, so there's more to do.


## QA steps

bootstrap k8s and deploy a charm
juju show-controller and check the api address is correct

bootstrap lxd and deploy a charm
juju show-controller and check the api address is correct
juju enable-ha
juju show-controller and check the api addresses are correct
remove a controller machine
enable-ha again
check that a replacement controller comes up and show-controller is happy

## Bug reference

https://bugs.launchpad.net/juju/+bug/1866643
